packages/blade/src/components/Input/BaseInput/StyledBaseInput.web.tsx

```diff
@@ -1,41 +1,65 @@
 import React from 'react';
 import styled from 'styled-components';
+import type { CSSObject, DefaultTheme, ThemeProps } from 'styled-components';
 import { getBaseInputStyles } from './baseInputStyles';
 
 import type { StyledBaseInputProps } from './types';
 import getTextStyles from '~components/Typography/Text/getTextStyles';
 
+const getWebInputStyles = (
+  props: Omit<StyledBaseInputProps, 'accessibilityProps' | 'setCurrentInteraction' | 'type'> &
+    ThemeProps<DefaultTheme> &
+    (React.InputHTMLAttributes<HTMLInputElement> | React.ButtonHTMLAttributes<HTMLButtonElement>),
+): CSSObject => {
+  return {
+    ...getBaseInputStyles({
+      isDisabled: props.disabled,
+      theme: props.theme,
+      validationState: props.validationState,
+      leadingIcon: props.leadingIcon,
+      prefix: props.prefix,
+      interactionElement: props.interactionElement,
+      suffix: props.suffix,
+      trailingIcon: props.trailingIcon,
+      textAlign: props.textAlign,
+    }),
+    outline: 'none',
+    border: 'none',
+    '::placeholder': {
+      ...getTextStyles({
+        size: 'medium',
+        variant: 'body',
+        type: 'placeholder',
+        weight: 'regular',
+        contrast: 'low',
+        theme: props.theme,
+      }),
+      textAlign: props.textAlign,
+    },
+    ':focus': {
+      outline: 'none',
+    },
+    cursor: props.disabled ? 'not-allowed' : 'auto',
+  };
+};
+
 const StyledBaseNativeInput = styled.input<
   Omit<StyledBaseInputProps, 'accessibilityProps' | 'setCurrentInteraction' | 'type'>
+>(getWebInputStyles);
+
+const StyledBaseNativeButton = styled.button<
+  Omit<StyledBaseInputProps, 'accessibilityProps' | 'setCurrentInteraction' | 'type'>
 >((props) => ({
-  ...getBaseInputStyles({
-    isDisabled: props.disabled,
+  ...getWebInputStyles(props),
+  ...getTextStyles({
```

Comment:
getTextStyles is already being called within `getBaseInputStyles` right? so couldn't understand why do we need to call it again here? 🤔

Reply:
So the same styles are applied on `::placeholder` as well as innerText. which is why I abstracted it out into this function and now I am calling it for `::placeholder` inside `getWebInputStyles` and adding it for innerText on Button's styles